lib/segment/src/types.rs

```diff
@@ -2025,12 +2025,31 @@ pub struct WithPayload {
     pub payload_selector: Option<PayloadSelector>,
 }
 
+#[derive(Debug, Deserialize, Serialize, JsonSchema, Validate, Clone, PartialEq, Default)]
+#[serde(deny_unknown_fields, rename_all = "snake_case")]
```

Comment:
Is `deny_unknown_fields` required? Usually we prefer not to use it unless it is strictly necessary. Otherwise it would be core complicated to extend this data structure further and maintain compatibility with previous versions.